algorithmic_efficiency/workloads/fastmri/fastmri_jax/input_pipeline.py

```diff
@@ -0,0 +1,256 @@
+  prob = (num_cols_float / acceleration - num_low_frequencies_float) / (
+      num_cols_float - num_low_frequencies_float)
+
+  mask = tf.cast(
+      tf.random.stateless_uniform((num_cols,), seed) < prob, dtype=tf.float32)
+  acceleration_mask = tf.reshape(mask, (1, num_cols))
+
+  mask = tf.math.maximum(center_mask, acceleration_mask)
+  mask = tf.cast(mask, dtype=tf.complex64)
+
+  # apply_mask
+  masked_kspace = kspace * mask + 0.0
+
+  # ifft2c
+  shifted_kspace = tf.signal.ifftshift(masked_kspace, axes=(0, 1))
+  shifted_image = tf.signal.ifft2d(shifted_kspace)
+  image = tf.signal.fftshift(shifted_image, axes=(0, 1))
+  scaling_norm = tf.cast(
+      tf.math.sqrt(
+          tf.cast(tf.math.reduce_prod(tf.shape(kspace)[-2:]), 'float32')),
+      kspace.dtype)
+  image = image * scaling_norm
+  image = tf.stack((tf.math.real(image), tf.math.imag(image)), axis=-1)
+
+  # complex_center_crop
+  w_from = (kspace_shape[0] - target_shape[0]) // 2
+  h_from = (kspace_shape[1] - target_shape[1]) // 2
+  w_to = w_from + target_shape[0]
+  h_to = h_from + target_shape[1]
+
+  image = image[..., w_from:w_to, h_from:h_to, :]
+
+  # complex_abs
+  abs_image = tf.math.sqrt(tf.math.reduce_sum(image**2, axis=-1))
+
+  # normalize_instance
+  mean = tf.math.reduce_mean(abs_image)
+  std = tf.math.reduce_std(abs_image)
+  norm_image = (abs_image - mean) / std
+
+  # clip_image
+  image = tf.clip_by_value(norm_image, -6, 6)
+
+  # process target
+  norm_target = (target - mean) / std
+  target = tf.clip_by_value(norm_target, -6, 6)
+
+  return {
+      'inputs': image,
+      'targets': target,
+      'mean': mean,
+      'std': std,
+      'volume_max': volume_max,
+      'weights': tf.ones((image.shape[0],)),
+  }
+
+
+def _h5_to_examples(path):
+  """Yield MRI slices from an hdf5 file containing a single MRI volume."""
+  tf.print('fastmri_dataset._h5_to_examples call:',
+           path,
+           datetime.datetime.now().strftime('%H:%M:%S:%f'))
+  with gfile.GFile(path, 'rb') as gf:
+    with h5py.File(gf, 'r') as hf:
+      # NOTE(dsuo): logic taken from reference code
+      volume_max = hf.attrs.get('max', 0.0)
+
+      for i in range(hf['kspace'].shape[0]):
+        yield hf['kspace'][i], hf['kspace'][i].shape, hf['reconstruction_esc'][
+            i], hf['reconstruction_esc'][i].shape, volume_max
+
+
+def _create_generator(filename):
+  signature = (
+      tf.TensorSpec(shape=(640, None), dtype=tf.complex64),
+      tf.TensorSpec(shape=(2,), dtype=tf.int32),
+      tf.TensorSpec(shape=(320, 320), dtype=tf.float32),
+      tf.TensorSpec(shape=(2,), dtype=tf.int32),
+      tf.TensorSpec(shape=(), dtype=tf.float32),
+  )
+  return tf.data.Dataset.from_generator(
+      _h5_to_examples, args=(filename,), output_signature=signature)
+
+
+def _pad_zeros_like(per_host_batch_size, x):
+  shape = x.shape[1:]
+  pad_size = per_host_batch_size - x.shape[0]
+  return tf.zeros((pad_size, *shape), x.dtype)
+
+
+def load_fastmri_split(per_host_batch_size,
```

Comment:
Unless I'm missing something, `per_host_batch_size` is the global batch size, since the batches are sharded with `data_utils.shard_numpy_ds()` below (see my other comments).